db/transaction_log_impl.cc

```diff
@@ -74,14 +74,21 @@ Status TransactionLogIteratorImpl::OpenLogFile(
   return s;
 }
 
-BatchResult TransactionLogIteratorImpl::GetBatch()  {
+BatchResult TransactionLogIteratorImpl::GetBatch() {
   assert(is_valid_);  //  cannot call in a non valid state.
+  assert(current_batch_);
   BatchResult result;
   result.sequence = current_batch_seq_;
   result.writeBatchPtr = std::move(current_batch_);
   return result;
 }
 
+void TransactionLogIteratorImpl::RecycleWriteBatch(
+    std::unique_ptr<WriteBatch> write_batch) {
+  assert(write_batch);
+  current_batch_ = std::move(write_batch);
```

Comment:
Isn't it better to clear the batch here?

Reply:
Adjusted the PR to clear there as well...